mage_ai/frontend/components/IntegrationPipeline/index.tsx

```diff
@@ -127,6 +132,13 @@ function IntegrationPipeline({
 
     return parse(dataLoaderBlock.content);
   }, [dataLoaderBlock]);
+
+  useEffect(() => {
+    if (dataLoaderBlockContent?.source) {
+      setIntegrationSourceDocs(integrationSourcesByUUID[dataLoaderBlockContent.source]?.docs);
+    }
+  }, [integrationSources, dataLoaderBlockContent]);
```

Comment:
`integrationSourcesByUUID` missing from dependency array (unless that was intentional to reduce re-renders)

Reply:
Oh good point, I will update the useEffect to depend on `integrationSourcesByUUID` instead of `integrationSources`, that makes sense.